### PR TITLE
include props and targets for code style packages

### LINKED ISSUE #22870: Analysis mode for code style analyzers does not work in .Net SDK 6
### Describe the bug

According to documentation at https://docs.microsoft.com/en-us/dotnet/core/project-sdk/msbuild-props we should be able to set analysis mode (e.g. Recommended) for __code style__ analyzers. This does not seem to work at all.

When talking about `AnalysisLevel` property, the docs say:
> In .NET 5 and earlier versions, this property only affects code-quality (CAXXXX) rules. Starting in .NET 6, if you set EnforceCodeStyleInBuild to true, this property affects code-style (IDEXXXX) rules too.

Later it talks about `AnalysisLevelStyle` property:
> AnalysisLevelStyle | All code-style (IDEXXXX) rules

Then later when talking about `AnalysisMode` property, the docs say again:
> In .NET 5, this property only affects code-quality (CAXXXX) rules. Starting in .NET 6, if you set EnforceCodeStyleInBuild to true, this property affects code-style (IDEXXXX) rules too.

None of the above mentioned properties seem to do anything. And I don't see how they could. There's no editor config file named like `analysislevelstyle_[...].editorconfig` under `6.0.100/Sdks/Microsoft.NET.Sdk/analyzers/build/config`. And there are no other editorconfig files at all outside of that location in the SDK.

Are the docs wrong and it's not actually supported or is something missing in the SDK?

### To Reproduce

Use VS2022, create a new Asp.Net Core Web project (project type is irrelevant). 
Open the project and add EnforceCodeStyleInBuild to enable style analyzers:
```
<PropertyGroup>
  <EnforceCodeStyleInBuild>true</EnforceCodeStyleInBuild>
</PropertyGroup>
```

Now try to enable `Recommended` mode:
1. Add `<AnalysisLevel>6-recommended</AnalysisLevel>` OR 
2. Add `<AnalysisMode>Recommended</AnalysisMode>` OR 
3. Add `<AnalysisLevelStyle>6-recommended</AnalysisLevelStyle>`

After each attempt you can observe in VS the rule severity under `Dependencies\Analyzers\Microsoft.CodeAnalysis.CSharp.CodeStyle` and see that nothing changes, nearly all rules are at `silent` level and a few at `suggestion`. 

### Further technical details
- Include the output of `dotnet --info`
- The IDE (VS / VS Code/ VS4Mac) you're running on, and its version

Using .Net SDK 6.0.100, VS 2022 17.0.1, on Windows.

### PR DESCRIPTION
**Pull request by jmarolf**
Fixes https://github.com/dotnet/sdk/issues/22870

Here is a view of what the directory structure in the SDK looks like with this change

```ps1
> tree /F artifacts/bin/Debug/Sdks/Microsoft.NET.Sdk/codestyle
C:\S\S\ARTIFACTS\BIN\DEBUG\SDKS\MICROSOFT.NET.SDK\CODESTYLE
├───cs
│   │   Microsoft.CodeAnalysis.CodeStyle.dll
│   │   Microsoft.CodeAnalysis.CodeStyle.Fixes.dll
│   │   Microsoft.CodeAnalysis.CSharp.CodeStyle.dll
│   │   Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.dll
│   │
│   ├───build
│   │   │   Microsoft.CodeAnalysis.CSharp.CodeStyle.targets
│   │   │
│   │   └───config
│   │           AnalysisLevelStyle_All.editorconfig
│   │           AnalysisLevelStyle_Default.editorconfig
│   │           AnalysisLevelStyle_Minimum.editorconfig
│   │           AnalysisLevelStyle_None.editorconfig
│   │           AnalysisLevelStyle_Recommended.editorconfig
│   │
│   ├───cs
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───de
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───es
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───fr
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───it
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───ja
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───ko
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───pl
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───pt-BR
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───ru
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───tr
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   ├───zh-Hans
│   │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│   │       Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│   │
│   └───zh-Hant
│           Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
│           Microsoft.CodeAnalysis.CodeStyle.resources.dll
│           Microsoft.CodeAnalysis.CSharp.CodeStyle.Fixes.resources.dll
│           Microsoft.CodeAnalysis.CSharp.CodeStyle.resources.dll
│
└───vb
    │   Microsoft.CodeAnalysis.CodeStyle.dll
    │   Microsoft.CodeAnalysis.CodeStyle.Fixes.dll
    │   Microsoft.CodeAnalysis.VisualBasic.CodeStyle.dll
    │   Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.dll
    │
    ├───build
    │   │   Microsoft.CodeAnalysis.VisualBasic.CodeStyle.targets
    │   │
    │   └───config
    │           AnalysisLevelStyle_All.editorconfig
    │           AnalysisLevelStyle_Default.editorconfig
    │           AnalysisLevelStyle_Minimum.editorconfig
    │           AnalysisLevelStyle_None.editorconfig
    │           AnalysisLevelStyle_Recommended.editorconfig
    │
    ├───cs
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───de
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───es
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───fr
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───it
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───ja
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───ko
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───pl
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───pt-BR
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───ru
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───tr
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    ├───zh-Hans
    │       Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.CodeStyle.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
    │       Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
    │
    └───zh-Hant
            Microsoft.CodeAnalysis.CodeStyle.Fixes.resources.dll
            Microsoft.CodeAnalysis.CodeStyle.resources.dll
            Microsoft.CodeAnalysis.VisualBasic.CodeStyle.Fixes.resources.dll
            Microsoft.CodeAnalysis.VisualBasic.CodeStyle.resources.dll
```

### CODE REVIEW

File: src/Layout/redist/targets/GenerateLayout.targets [hunk 1]
```diff
@@ -35,34 +35,51 @@
...
       <CodeStyleVisualBasicAssemblies Include="$(PkgMicrosoft_CodeAnalysis_VisualBasic_CodeStyle)/analyzers/dotnet/vb/**/*.dll" />
-      <ILLinkAnalyzersAssemblies Include="$(PkgMicrosoft_NET_ILLink_Analyzers)/analyzers/dotnet/cs/**/*.dll" />
-      <AnalyzerConfig Include="$(PkgMicrosoft_CodeAnalysis_NetAnalyzers)/build/config/**/*" />
+      <CodeStyleTargets Include="$(PkgMicrosoft_CodeAnalysis_CSharp_CodeStyle)/build/Microsoft.CodeAnalysis.CSharp.CodeStyle.props" />
+      <CodeStyleTargets Include="$(PkgMicrosoft_CodeAnalysis_CSharp_CodeStyle)/build/Microsoft.CodeAnalysis.CSharp.CodeStyle.targets" />
+      <CodeStyleTargets Include="$(PkgMicrosoft_CodeAnalysis_VisualBasic_CodeStyle)/build/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.props" />
+      <CodeStyleTargets Include="$(PkgMicrosoft_CodeAnalysis_VisualBasic_CodeStyle)/build/Microsoft.CodeAnalysis.VisualBasic.CodeStyle.targets" />
+      <!-- The files in the build/config folder are identical across packages -->
+      <CodeStyleConfig Include="$(PkgMicrosoft_CodeAnalysis_CSharp_CodeStyle)/build/config/**/*" />
```
Comment (mavasani) [inline src/Layout/redist/targets/GenerateLayout.targets:62]: This should also have a corresponding VB include?

Comment (mavasani) [reply src/Layout/redist/targets/GenerateLayout.targets]: Ah, I didn’t realize that. Can you add a comment on the same lines?

**Review by mavasani — APPROVED** [review]

**Comment by jmarolf** [comment]
As per the comment here: https://github.com/dotnet/roslyn/blob/9d335cd405b29f7ecfe7b8a9a716176de897c62f/src/CodeStyle/CSharp/CodeFixes/build/Microsoft.CodeAnalysis.CSharp.CodeStyle.props#L3-L8
> This props file should only be present in the analyzer NuGet package, it should **not** be inserted into the .NET SDK.

I am not including the props files :)

**Comment by jmarolf** [comment]
After extensively investigating all the test failures they appear to all be false positives. I am re-basing and re-running the tests

**Pull request merged**